domehunter/gRPC-server/hx2dome.proto_server.py

```diff
@@ -0,0 +1,578 @@
+# dapiFindHome            (google.protobuf.Empty)   returns   (ReturnCode) {};
+# dapiIsGotoComplete      (google.protobuf.Empty)   returns   (IsComplete) {};
+# dapiIsOpenComplete      (google.protobuf.Empty)   returns   (IsComplete) {};
+# dapiIsCloseComplete     (google.protobuf.Empty)   returns   (IsComplete) {};
+# dapiIsParkComplete      (google.protobuf.Empty)   returns   (IsComplete) {};
+# dapiIsUnparkComplete    (google.protobuf.Empty)   returns   (IsComplete) {};
+# dapiIsFindHomeComplete  (google.protobuf.Empty)   returns   (IsComplete) {};
+# dapiSync                (AzEl)                    returns   (ReturnCode) {};
+
+
+class HX2DomeServer(hx2dome_pb2_grpc.HX2DomeServicer):
+    """Remote Procedure Call (RPC) server object.
+
+    Parameters
+    ----------
+    testing : bool
+        Parameter to toggle testing (simulated hardware) mode.
+    debug_lights : bool
+        Parameter to toggle automationHAT status LEDs (automationHAT required).
+    server_testing : bool
+        Parameter to toggle server testing mode, to test communication between
+        TheSkyX and the RPC server without needing any real or simulated
+        hardware.
+
+    Attributes
+    ----------
+    dome : Huntsman Dome object
+        This the object that represents the Huntsman Dome as controlled via a
+        raspberryPi and an automationHAT. It can also be initialised in a
+        testing mode with simulated hardware.
+
+    """
+
+    def __init__(self, testing, debug_lights, server_testing):
+        super(HX2DomeServer, self).__init__()
+        # create the dome object that controls the dome hardware
+        self.dome = Dome(testing=testing, debug_lights=debug_lights)
+        self.server_testing = server_testing
+
+    def dapiGetAzEl(self, request, context):
+        """TheSkyX RPC to query the dome azimuth and slit position of the
+        observatory.
+
+        Parameters
+        ----------
+        request : Empty
+            Incoming rpc request, a message of type 'Empty'.
+        context : Dunno
+            GRPC magic thingy.
+
+        Returns
+        -------
+        AzEl
+            rpc response, a message of type 'AzEl' describing the dome's
+            azimuth and slit elevation postions.
+
+        """
+        print(f'Receiving: GetAzEl request')
+        # our cpp TheSkyX driver uses the dapiGetAzEl rpc to check link
+        if self.server_testing:
+            # if we just want to test communication between driver and server
+            # we can just send back a dummy response
+            response = hx2dome_pb2.AzEl(return_code=0, az=10.0, el=20.0)
+            print(f'Sending: Az={response.az}, El={response.el}')
+            return response
+        else:
+            return_code = 0
+            try:
+                dome_az = self.dome.getAz()
+            except:
+                # TODO: proper exception handling
+                dome_az = None
+            if dome_az is None:
+                return_code = 1
```

Comment:
Could just put this in the `except Exception` block rather than have a separate `if` statement.